File: solutions/c/raindrops/3/raindrops.c
#include "raindrops.h"
#include <string.h>
#include <stdio.h>

void convert(char result[], int drops){  

    result[0] = '\0';

    //If divisible by 3
    if (drops % 3 == 0) {
        strcat(result, "Pling");   
    }
    //If divisible by 5
    if (drops % 5 == 0) {
        strcat(result, "Plang"); 
    }
    //If divisible by 7
    if (drops % 7 == 0) {
        strcat(result, "Plong");
    }
    //If not divisible by 3, 5 or 7 length of the string would be 0
    if (strlen(result) == 0) { 
        //Inserts drops as string into the result array.
        sprintf(result, "%d", drops); 
        
    }    
}

File: solutions/python/raindrops/3/raindrops.py
def convert(number):
    result = ""
    divisibility = (number % 3 == 0, number % 5 == 0, number % 7 == 0)
    sounds = ("Pling", "Plang", "Plong")
    result = "".join(sound for is_div, sound in zip(divisibility, sounds) if is_div)
    return result or str(number)